minerl/data/util/blacklist.py

```diff
@@ -1,14 +1,19 @@
 import os
+import pathlib
 from minerl.data.util.constants import touch
 
+MODULE_PATH = pathlib.Path(__file__).parent.absolute()
+BLACKLIST_DIR_PATH = MODULE_PATH / 'assets' / 'blacklist'
+
 
 class Blacklist:
 
-    def __init__(self):
-        self.file_name = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'assets', 'blacklist'))
-        os.makedirs(self.file_name, exist_ok=True)
-    def add(self, other):
-        touch(os.path.join(self.file_name, other))
+    def __init__(self, blacklist_dir=BLACKLIST_DIR_PATH):
+        self.blacklist_dir = blacklist_dir
+        os.makedirs(self.blacklist_dir, exist_ok=True)
+
+    def add(self, entry):
+        touch(self.blacklist_dir / entry)
 
     def __contains__(self, item):
-        return item in os.listdir(self.file_name)
\ No newline at end of file
+        return item in os.listdir(self.blacklist_dir)
```

Comment:
Do we need to also move files around to preserve their blacklist? Probably not necessary since we should only care about our own experiments, but they might want that if we PR back in.

Reply:
The nice thing about using a filesystem of empty files to maintain a blacklist is that synchronization is automatically solved between parallel workers in the pipeline. We could switch to something else, but that would use up more engineering time and could have annoying synchronization issues that would be a pain to test.